pkg/fedresolver/fedschema.go

```diff
@@ -0,0 +1,175 @@
+package fedresolver
+
+import (
+	"context"
+	"crypto/tls"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"strings"
+	"sync"
+	"time"
+
+	"k8s.io/klog/v2"
+)
+
+// Define the httpClient interface
+type httpClient interface {
+	Do(req *http.Request) (*http.Response, error)
+}
+
+// clientWrapper is a generic wrapper around http.Client or any httpClient implementation
+type clientWrapper struct {
+	client httpClient
+}
+
+func (c *clientWrapper) Do(req *http.Request) (*http.Response, error) {
+	return c.client.Do(req)
+}
+
+// Your SearchSchema struct with an interface for the HTTP client
+type SearchSchema struct {
+	routeToken []map[string]string
+	client     httpClient
+}
+
+type schemaPayload struct {
+	Data struct {
+		SearchSchema map[string][]string `json:"searchSchema"`
+	} `json:"data"`
+}
+
+func SearchSchemaResolver(ctx context.Context) (map[string]interface{}, error) {
+	routeToken := getTokenAndRoute()
+
+	// Create an http.Client
+	httpClient := &http.Client{
+		Transport: &http.Transport{
+			TLSClientConfig: &tls.Config{InsecureSkipVerify: true},
```

Comment:
It's ok to skip TLS verification for now, but let's add something to remember that we need to revisit later.